lib/internal/streams/from.js

```diff
@@ -73,15 +74,20 @@ function from(Readable, iterable, opts) {
       needToClose = false;
       const { value, done } = await iterator.next();
       needToClose = !done;
-      const resolved = await value;
       if (done) {
         readable.push(null);
       } else if (readable.destroyed) {
         await close();
-      } else if (readable.push(resolved)) {
-        next();
       } else {
-        reading = false;
+        const res = await value;
+        if (res === null) {
+          reading = false;
+          readable.destroy(new ERR_STREAM_NULL_VALUES());
```

Comment:
@lpinca: throw or destroy?

Reply:
This is identical, no? If it's thrown, it would end up in the catch clause and then call `readable.destroy(error)`? Keeping it as is seems fine therefore.

Reply:
You are correct. I forgot about the catch clause.